extensions/typescript-language-features/src/languageFeatures/tsconfig.ts

```diff
@@ -110,6 +114,55 @@ class TsconfigLinkProvider implements vscode.DocumentLinkProvider {
 	}
 }
 
+// Reference: https://github.com/microsoft/TypeScript/blob/febfd442cdba343771f478cf433b0892f213ad2f/src/compiler/commandLineParser.ts#L3005
+/**
+ * @returns Returns undefined in case of lack of result while trying to resolve from node_modules
+ */
+const getTsconfigPath = async (baseDirUri: vscode.Uri, extendsValue: string): Promise<vscode.Uri | undefined> => {
+	// Don't take into account a case, where tsconfig might be resolved from the root (see the reference)
+	// e.g. C:/projects/shared-tsconfig/tsconfig.json (note that C: prefix is optional)
+
+	const isRelativePath = ['./', '../'].some(str => extendsValue.startsWith(str));
+	if (isRelativePath) {
+		const absolutePath = vscode.Uri.joinPath(baseDirUri, extendsValue);
+		if (await exists(absolutePath)) { return absolutePath; }
+		// Will suggest to create a .json variant if it doesn't exist yet
+		return absolutePath.with({
+			path: `${absolutePath.path}.json`
+		});
+	}
+
+	// Otherwise resolve like a module
+	return resolveNodeModulesPath([
+		extendsValue,
+		`${extendsValue}.json`,
+		`${extendsValue}/tsconfig.json`,
+	]);
+
+	async function resolveNodeModulesPath(pathCandidates: string[]): Promise<vscode.Uri | undefined> {
```

Comment:
[nit] Not a fan of nesting functions definitions like this. Instead move this to the top level of the file and pass in any needed parameters

Reply:
I firstly resolved it because I thought you was talking about moving it above the `return ...` statement. But whats the point of having it defined on top level, since it's being used only in one place? I'm pretty sure this approach reduces readability of your codebase.

TLDR By keeping the definition scoped it is much easier to understand where it's being used. WDYT?

Reply:
Didn't see a reply so I just moved it like you said. I don't mind of almost any codestyle, sorry for arguing. Btw what nit does mean?